File: objectiveai-js/src/functions/executions/response/streaming/object.ts
import { z } from "zod";

export const FunctionsExecutionsResponseStreamingObjectSchema = z.enum(["scalar.function.execution.chunk","vector.function.execution.chunk"]).meta({ title: "functions.executions.response.streaming.Object" });
export type FunctionsExecutionsResponseStreamingObject = z.infer<typeof FunctionsExecutionsResponseStreamingObjectSchema>;

File: objectiveai-rs/src/functions/executions/response/streaming/object.rs
use serde::{Deserialize, Serialize};
use schemars::JsonSchema;

#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize, JsonSchema)]
#[schemars(rename = "functions.executions.response.streaming.Object")]
pub enum Object {
    #[serde(rename = "scalar.function.execution.chunk")]
    ScalarFunctionExecutionChunk,
    #[serde(rename = "vector.function.execution.chunk")]
    VectorFunctionExecutionChunk,
}
